text/0000-buildpack-docs.md

```diff
@@ -0,0 +1,66 @@
+# Meta
+[meta]: #meta
+- Name: Buildpack.toml documentation
+- Start Date: (fill in today's date: 2021-04-26)
+- Author(s): (dwillist)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: "N/A"
+
+# Summary
+[summary]: #summary
+We propose adding a `documentation` file `README.md` adjacent to the `buildpack.toml` file in buildpacks.
+
+We would also like a new OCI label `io.buildpacks.buildpack.docs` in order to attach documentation on the outside of builder images and buildpackages.
+
+
+# Motivation
+[motivation]: #motivation
+
+- Why should we do this?
+  - To keep documentation accessible from the pack CLI & other buildpack tooling.
+  - Establish common locations for docs & promote similar documentation strategies between buildpacks.
+
+- What is the expected outcome?
+    - buildpacks have a standard location for documentation
+    - users can more easily figure out how groups of buildpacks work together.
+
+# What it is
+[what-it-is]: #what-it-is
+Just adding a new optional `README.md` adjacent to the `buildpack.toml` file in buildpacks and a new OCI label `io.buildpacks.buildpack.docs` on buildpackages and builder images.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+Just adding the above file & image label would allow for `pack` to add some command exposing documentation when inspecting builders & any buildpack locator.
+
+This documentation could then be easily displayed by `pack inspect ...` commands as well as used to add some more content to the registry.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Polluting the image labels.
+
+Really would be nice to have a single documentation solution for folks that already have nice HTML docs see the [paketo project](https://paketo.io/docs/buildpacks/language-family-buildpacks/go/)
+
+# Alternatives
+[alternatives]: #alternatives
+
+Could add documentation directly into the `buildpack.toml` file under a `documentation` key.
+
+Could ditch all documentation at the command line level & just add documentation to the registry when running pack register.
+
```

Comment:
If we were going to include documentation on the images themselves, in order to make it more useful than the documentation URLs I would prefer to see something more structured so that `pack` could do thing like print all environment variables that could be used to configure a build or seperately to configure the app at runtime.

Some prior art: Paketo Java buildpacks docoument their configuration options in `buildpack.toml` and print a table describing these options during the build. The build logs are probably not the best place to surface this information, but it's the only options that exists right now.